pkg/identityserver/user_registry_test.go

```diff
@@ -122,13 +123,72 @@ func TestUserCreate(t *testing.T) {
 			a.So(errors.Resemble(err, errUserRegistrationDisabled), should.BeTrue)
 		}
 
-		is.config.UserRegistration.Enabled = true
-		is.config.UserRegistration.Invitation.Required = true
-
-		_, err = reg.Create(ctx, req)
-		if a.So(err, should.NotBeNil) {
-			a.So(errors.Resemble(err, errInvitationTokenRequired), should.BeTrue)
-		}
+		t.Run("InvitationRequired", func(t *testing.T) {
+			is.config.UserRegistration.Enabled = true
```

Comment:
I find it wrong that you are moving the config update inside the test - it will affect future tests too, so it should stay outside - it makes it look like this is affecting only this test, while it does not.